be/src/exprs/string_functions.h

```diff
@@ -570,6 +601,54 @@ class StringFunctions {
         return ss.str();
     };
 
+    /**
+    * FORMAT(s,d)
+    * Formats the number string s to a format like '#,###,###.##', truncated to d decimal places, and returns the result as a string.
+    * If d is 0, the result has no decimal point or fractional part; If d is negative, treat like 0.
+    * Rounding should be handled before entering this method.
+    * FORMAT(12332.123456, 4) -> "12,332.1234"
+    * FORMAT(12332.2,0) -> "12,332"
+    * FORMAT(12332.1,4) -> "12,332.1000"
+    * FORMAT(1.1, -4) -> "1"
+    */
+    static std::string transform_format(const std::string& s, int d) {
```

Comment:
can we achieve this  by using standard library? for example, format double by this way
```cpp
    std::ostringstream oss;
    oss.imbue(std::locale("en_US.UTF-8"));
    oss << std::fixed << std::setprecision(3)  << (double)(123456.8901);
```